### PR TITLE
[enhance](cooldown) turn write cooldown meta async

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #xxx
Upload meta file first once this replica is chosen to be new cooldown replica.
Consider such case:
replica A: term 15, replica B: term 14, replica C: term 15
and the former cooldown replica is A
if FE chooses C to be the new cooldown replica, and B tries to
follow C now, B wouldn't be able to find .meta file on remote storage
## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.29 seconds
 stream load tsv:          468 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         35 seconds loaded 2358488459 Bytes, about 64 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230216055503_clickbench_pr_97532.html

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -1772,11 +1796,31 @@ Status check_version_continuity(const std::vector<RowsetMetaSharedPtr>& rs_metas
     return Status::OK();
 }
 
-Status Tablet::write_cooldown_meta() {
+// It's guaranteed the write cooldown meta task would be invoked at the end unless BE crashes
+// one tablet would at most have one async task to be done
+void Tablet::async_write_cooldown_meta() {
+    WriteCooldownMetaExecutors::GetInstance()->submit(tablet_id(), [this]() {
```
Comment (platoneko) [inline be/src/olap/tablet.cpp:1802]: `this` seems unsafe

Comment (platoneko) [reply be/src/olap/tablet.cpp:1802]: Seems that it's not a good idea to provide `Tablet::async_write_cooldown_meta`, just make a closure and capture tablet's copied shared ptr, call `tablet->write_cooldown_meta` in this closure.

Comment (platoneko) [reply be/src/olap/tablet.cpp:1802]: If `write_cooldown_meta` returns not ok and not aborted, enqueue this tablet to executor again.

**Comment by ByteYue** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**